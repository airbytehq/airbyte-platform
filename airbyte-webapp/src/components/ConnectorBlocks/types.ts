export enum StepsTypes {
  OVERVIEW = "overview",
  SETTINGS = "settings",
}
