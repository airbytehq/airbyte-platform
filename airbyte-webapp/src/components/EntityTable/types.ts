import {
  ActorDefinitionVersionBreakingChanges,
  ActorDefinitionVersionRead,
  ConnectionScheduleData,
  ConnectionScheduleType,
  SchemaChange,
  SupportState,
  WebBackendConnectionListItem,
} from "../../core/api/types/AirbyteClient";

interface EntityTableDataItem {
  entityId: string;
  entityName: string;
  connectorName: string;
  connectEntities: Array<{
    name: string;
    connector: string;
    status: string;
    lastSyncStatus: string | null;
  }>;
  enabled: boolean;
  lastSync?: number | null;
  connectorIcon?: string;
  isActive: boolean;
  breakingChanges?: ActorDefinitionVersionBreakingChanges;
  isVersionOverrideApplied: boolean;
  supportState?: SupportState;
}

interface ConnectionTableDataItem {
  connectionId: string;
  name: string;
  entityName: string;
  connectorName: string;
  enabled: boolean;
  isSyncing?: boolean;
  status?: string;
  lastSync?: number | null;
  scheduleData?: ConnectionScheduleData;
  scheduleType?: ConnectionScheduleType;
  schemaChange: SchemaChange;
  source: ActorDefinitionVersionRead;
  destination: ActorDefinitionVersionRead;
  lastSyncStatus: string | null;
  connectorIcon?: string;
  entityIcon?: string;
  connection: WebBackendConnectionListItem;
}

const enum Status {
  ACTIVE = "active",
  INACTIVE = "inactive",
  FAILED = "failed",
  CANCELLED = "cancelled",
  EMPTY = "empty",
  PENDING = "pending",
}

enum SortOrderEnum {
  DESC = "desc",
  ASC = "asc",
}

export type { ConnectionTableDataItem, EntityTableDataItem };
export { Status, SortOrderEnum };
