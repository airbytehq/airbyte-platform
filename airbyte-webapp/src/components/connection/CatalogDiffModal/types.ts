import { FieldTransform, StreamTransform } from "core/api/types/AirbyteClient";

export type DiffVerb = "new" | "removed" | "changed";

export interface SortedDiff<T extends StreamTransform | FieldTransform> {
  newItems: T[];
  removedItems: T[];
  changedItems: T[];
}
