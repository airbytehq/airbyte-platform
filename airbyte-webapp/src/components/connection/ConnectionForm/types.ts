import { NamespaceDefinitionType } from "core/request/AirbyteClient";

export const namespaceDefinitionOptions: Record<NamespaceDefinitionType, string> = {
  [NamespaceDefinitionType.destination]: "destinationFormat",
  [NamespaceDefinitionType.source]: "sourceFormat",
  [NamespaceDefinitionType.customformat]: "customFormat",
};
