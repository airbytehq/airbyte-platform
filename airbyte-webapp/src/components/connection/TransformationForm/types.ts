import { OperationId, OperatorDbt } from "core/api/types/AirbyteClient";

export interface DbtOperationRead {
  name: string;
  workspaceId: string;
  operationId: OperationId;
  operatorConfiguration: {
    operatorType: "dbt";
    dbt: OperatorDbt;
  };
}

export interface DbtOperationReadOrCreate extends Omit<DbtOperationRead, "operationId"> {
  operationId?: OperationId;
}
