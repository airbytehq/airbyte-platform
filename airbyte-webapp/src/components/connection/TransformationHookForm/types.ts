import { OperationId, OperatorDbt } from "core/request/AirbyteClient";

export interface DbtOperationRead {
  name: string;
  workspaceId: string;
  operationId: OperationId;
  operatorConfiguration: {
    operatorType: "dbt";
    dbt: OperatorDbt;
  };
}

export interface DbtOperationReadOrCreate extends Omit<DbtOperationRead, "operationId"> {
  operationId?: OperationId;
}
