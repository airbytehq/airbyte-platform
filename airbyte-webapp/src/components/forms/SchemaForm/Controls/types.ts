import { ReactElement } from "react";

import { AirbyteJsonSchema } from "../utils";

export interface BaseControlProps {
  name: string;
  label?: string;
  labelTooltip?: ReactElement;
  optional: boolean;
  header?: ReactElement;
  containerControlClassName?: string;
  onlyShowErrorIfTouched?: boolean;
  placeholder?: string;
  "data-field-path"?: string;
  disabled?: boolean;
}

export type OverrideByPath = Record<string, ReactElement | null>;

export interface BaseControlComponentProps {
  fieldSchema: AirbyteJsonSchema;
  baseProps: BaseControlProps;
  overrideByPath?: OverrideByPath;
  skipRenderedPathRegistration?: boolean;
  hideBorder?: boolean;
  nonAdvancedFields?: string[];
}
