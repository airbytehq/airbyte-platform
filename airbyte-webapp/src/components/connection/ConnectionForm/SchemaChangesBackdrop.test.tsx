import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { mockConnection, TestWrapper } from "test-utils/testutils";

import { SchemaChange } from "core/api/types/AirbyteClient";
import { FeatureItem } from "core/services/features";
const mockUseConnectionEditService = jest.fn();

jest.doMock("hooks/services/ConnectionEdit/ConnectionEditService", () => ({
  useConnectionEditService: mockUseConnectionEditService,
}));

const TestWrapperWithAutoDetectSchema: React.FC<React.PropsWithChildren<Record<string, unknown>>> = ({ children }) => (
  <TestWrapper features={[FeatureItem.AllowAutoDetectSchema]}>{children}</TestWrapper>
);

const buttonSpy = jest.fn();

const renderComponent = () =>
  render(
    <SchemaChangeBackdrop>
      <button data-testid="bg-button" onClick={() => buttonSpy}>
        don't click
      </button>
    </SchemaChangeBackdrop>,
    { wrapper: TestWrapperWithAutoDetectSchema }
  );

// eslint-disable-next-line @typescript-eslint/no-var-requires
const { SchemaChangeBackdrop } = require("./SchemaChangeBackdrop");

describe("SchemaChangesBackdrop", () => {
  it("renders with breaking changes and prevents background interaction", async () => {
    mockUseConnectionEditService.mockReturnValue({
      connection: { mockConnection, schemaChange: SchemaChange.breaking },
      schemaHasBeenRefreshed: false,
      schemaRefreshing: false,
    });

    const { getByTestId } = renderComponent();

    expect(getByTestId("schemaChangesBackdrop")).toMatchSnapshot();
    await userEvent.click(getByTestId("bg-button"));
    expect(buttonSpy).not.toHaveBeenCalled();
  });

  it("does not render if there are non-breaking changes", async () => {
    mockUseConnectionEditService.mockReturnValue({
      connection: { mockConnection, schemaChange: SchemaChange.non_breaking },
      schemaHasBeenRefreshed: false,
      schemaRefreshing: false,
    });

    const { queryByTestId, getByTestId } = renderComponent();

    expect(queryByTestId("schemaChangesBackdrop")).toBeFalsy();

    await userEvent.click(getByTestId("bg-button"));
    expect(buttonSpy).not.toHaveBeenCalled();
  });

  it("does not render if there are no changes", async () => {
    mockUseConnectionEditService.mockReturnValue({
      connection: { mockConnection, schemaChange: SchemaChange.no_change },
      schemaHasBeenRefreshed: false,
      schemaRefreshing: false,
    });

    const { queryByTestId, getByTestId } = renderComponent();

    expect(queryByTestId("schemaChangesBackdrop")).toBeFalsy();

    await userEvent.click(getByTestId("bg-button"));
    expect(buttonSpy).not.toHaveBeenCalled();
  });

  it("does not render if schema has been refreshed", async () => {
    mockUseConnectionEditService.mockReturnValue({
      connection: mockConnection,
      schemaHasBeenRefreshed: true,
      schemaRefreshing: false,
    });

    const { queryByTestId, getByTestId } = renderComponent();
    expect(queryByTestId("schemaChangesBackdrop")).toBeFalsy();

    await userEvent.click(getByTestId("bg-button"));
    expect(buttonSpy).not.toHaveBeenCalled();
  });
});
