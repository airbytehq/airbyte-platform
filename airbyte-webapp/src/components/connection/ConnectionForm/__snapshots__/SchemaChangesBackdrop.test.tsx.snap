// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchemaChangesBackdrop renders with breaking changes and prevents background interaction 1`] = `
<div
  class="schemaBreakingChanges"
  data-testid="schemaChangesBackdrop"
>
  <div
    class="directionRow gapMd alignItemsCenter justifyContentCenter wrapNowrap container schemaBreakingChanges__backdrop"
  >
    <div
      class="directionColumn gapLg alignItemsCenter justifyContentCenter wrapNowrap container schemaBreakingChanges__message"
    >
      <p
        class="text lg darkBlue center"
        data-type="text"
      >
        Please review the schema updates before making changes to the connection
      </p>
      <button
        class="button sizeXS typePrimaryDark"
        type="button"
      >
        <span
          class="childrenContainer"
        >
          Review changes
        </span>
      </button>
    </div>
  </div>
  <button
    data-testid="bg-button"
  >
    don't click
  </button>
</div>
`;
